packages/e2e-tests-playwright/specs/widgets/customizing-widgets.spec.js

```diff
@@ -0,0 +1,660 @@
+
+		await page.click( 'role=menuitemradio[name="Footer #2"]' );
+
+		// Should switch to and expand Footer #2.
+		await expect(
+			page.locator(
+				'role=heading[name="Customizing ▸ Widgets Footer #2"]'
+			)
+		).toBeVisible();
+
+		// The paragraph block should be moved to the new sidebar and have focus.
+		const movedParagraphBlock = page.locator(
+			'*role=document[name="Paragraph block"] >> text="First Paragraph"'
+		);
+		await expect( movedParagraphBlock ).toBeVisible();
+		await expect( movedParagraphBlock ).toBeFocused();
+	} );
+
+	test( 'should not render Block Settings sections', async ( { page } ) => {
+		// We add Block Settings as a section, but it shouldn't display to
+		// the user as a section on the main menu. It's simply how we
+		// integrate the G sidebar inside the customizer.
+		await expect(
+			page.locator( 'role=heading[name=/Block Settings/][level=3]' )
+		).not.toBeVisible();
+	} );
+
+	test( 'should stay in block settings after making a change in that area', async ( {
+		page,
+		testUtils,
+		widgetsCustomizerPage,
+	} ) => {
+		// Open footer block widgets
+		await widgetsCustomizerPage.expandWidgetArea( 'Footer #1' );
+
+		// Add a block to make the publish button active.
+		await widgetsCustomizerPage.addBlock( 'Paragraph' );
+		await page.keyboard.type( 'First Paragraph' );
+
+		await widgetsCustomizerPage.waitForPreviewIframe();
+
+		// Click Publish
+		await Promise.all( [
+			page.waitForResponse(
+				testUtils.createURL( '/wp-admin/admin-ajax.php' )
+			),
+			page.click( 'role=button[name="Publish"]' ),
+		] );
+		// Wait for publishing to finish.
+		await expect(
+			page.locator( 'role=button[name="Published"]' )
+		).toBeDisabled();
+
+		// Select the paragraph block
+		await page.focus( 'role=document[name="Paragraph block"]' );
+
+		// Click the three dots button, then click "Show More Settings".
+		await testUtils.showBlockToolbar();
+		await testUtils.clickBlockToolbarButton( 'Options' );
+		await page.click( 'role=menuitem[name="Show more settings"]' );
+
+		// Change `drop cap` (Any change made in this section is sufficient; not required to be `drop cap`).
+		await page.click(
+			'css=.typography-block-support-panel >> role=button[name=/^View( and add)? options$/]'
+		);
+		await page.click( 'role=menuitemcheckbox[name="Show Drop cap"]' );
+
+		await page.click( 'role=checkbox[name="Drop cap"]' );
+
+		// Now that we've made a change:
+		// (1) Publish button should be active
+		// (2) We should still be in the "Block Settings" area
+		await expect(
+			page.locator( 'role=button[name="Publish"]' )
+		).not.toBeDisabled();
+
+		// This fails on 539cea09 and earlier; we get kicked back to the widgets area.
+		// We expect to stay in block settings.
+		await expect(
+			page.locator(
+				'role=heading[name="Customizing ▸ Widgets ▸ Footer #1 Block Settings"][level=3]'
+			)
+		).toBeVisible();
+	} );
+} );
+
+class WidgetsCustomizerPage {
```

Comment:
This is a [Page Object Model](https://playwright.dev/docs/test-pom) we use in this test. We can also move this to `e2e-test-utils` if there are needs.